src/engine_core/src/input/definitions/MouseData.hpp

```diff
@@ -0,0 +1,19 @@
+/*! \file MouseData.hpp
+    \author Kyn21kx
+    \date 2024-03-06
+    \brief Structure that holds the current data of the mouse, as interpreted by SDL events
+*/
+
+#pragma once
+#include "KeyStates.hpp"
+#include "MouseButton.hpp"
+#include <map>
+
+struct MouseData
+{
+    std::map<EMouseButton, EKeyState> mouseButtonMap{};
```

Comment:
I think that std unordered_map might work better. however, we should benchmark it. 🤔